### DISCUSSION TITLE
Windows Defender detects Nuclei as Trojan:Script/Wacatac.H!ml

### DISCUSSION THREAD

**Discussion by testerina** [projectdiscovery/nuclei#3382] [Q&A] [answered]
![Screenshot_1](https://user-images.githubusercontent.com/96467195/213995295-bbe5539c-40c4-41de-b4e0-5f1bb8cba3ac.png)

**Comment by ArwynFr** [comment]
Works fine with 2.7.7, fails for all versions 2.7.8 and above

**Comment by jimen0** [comment]
Does this still happen with the most recent release? It seems there are no detections for the ZIP nor the exe in VirusTotal.

* ZIP: https://www.virustotal.com/gui/file/4024e624fed3d8896598cdb3872dd8cefef3989b8972bed4ef35a68c380e2405
* nuclei.exe: https://www.virustotal.com/gui/file/ff8a3f39c340819679dc66c2bbb94b58a90fe68badbe7fe2b4946fb04c671827

Can you please check with v2.8.8 @testerina?

**Comment by testerina** [comment]
@jimen0 yes it's detected.

![Screenshot_13](https://user-images.githubusercontent.com/96467195/215487262-256efc1e-628f-49dc-9bf9-f3cf49f89454.png)

**Comment by tarunKoyalwar** [comment]
@testerina  we are not able to reproduce this ? can you share more details 
Ex:
- system details (windows version etc)
- steps to reproduce
- does it happen if nuclei is installed via `pdtm` app
- does it happen if nuclei is installed using `go install xxx`
and/or anything else that can help us to investigate issue ?

**Comment by toufik-airane** [comment, marked as answer]
Hello everyone 👋 

**Context**
A recent issue was reported by @testerina regarding the [nuclei.exe pre-compiled binaries](https://github.com/projectdiscovery/nuclei/releases) being flagged as malware by Windows Defender.

**Investigation**
In order to investigate this matter, I conducted a scan of the nuclei binaries on [VirusTotal](https://www.virustotal.com/) for windows 32/64 from version 2.8.7 to 2.8.9.

The following table shows the results of the investigation.

| name | hash | score | analysis |
|---|---|---|---|
| nuclei_2.8.7_windows_386.exe | 27fc5f5a143515b883a880cea4282e4a2148da7aa0726dd1f10512734a3b112f | 1 / 69 | **Palo Alto Networks**: Generic.ml |
| nuclei_2.8.7_windows_386.zip | 95517893e724b36df4e3245cd7f0002646dd8cb097cda7c7158363697cdd4026 | 0 / 64 | none |
| nuclei_2.8.7_windows_amd64.exe | f1e1c9396bbb0728accd7b13990c7b3ed5ae4787cadd3075124efb2b8410b794 | 0 / 68 | none |
| nuclei_2.8.7_windows_amd64.zip | 4a29a004f83b85f42944b0cec8aef38b4c9c623c2d698337883bdcc8b697e8b0 | 2 / 63 | **Jiangmin**: Backdoor.Wizard.j<br />**Kaspersky**:Trojan.Win32.WildNeutron.gfy |
| nuclei_2.8.8_windows_386.exe | 3b2d245ca4014acdbe05058bffa33b7b1a8e32fc571f1b3111fdead385ca8b1a | 4 / 69 | **Gridinsoft (no cloud)**: Trojan.Win32.Miner.sa<br />**Kaspersky**: Not-a-virus:HEUR:RiskTool.Win64.Miner.gen<br />**Rising**: Hacktool.Miner!8.13A68 (CLOUD)<br />**ZoneAlarm by Check Point**: Not-a-virus:HEUR:RiskTool.Win64.Miner.gen |
| nuclei_2.8.8_windows_386.zip | 47f9ed1057369e273d53c4268ce73bc339c92821199f7013707cf430b52141cb | 5 / 61 | **Gridinsoft (no cloud)**: Trojan.Win32.Miner.sa<br />**Kaspersky**: Not-a-virus:UDS:RiskTool.Win64.Miner.gen<br />**MaxSecure**: Trojan.Malware.74285590.susgen<br />**Rising**: Hacktool.Miner!8.13A68 (CLOUD)<br />**ZoneAlarm by Check Point**: Not-a-virus:HEUR:RiskTool.Win64.Miner.gen |
| nuclei_2.8.8_windows_amd64.exe | ff8a3f39c340819679dc66c2bbb94b58a90fe68badbe7fe2b4946fb04c671827 | 5 / 67 | **Alibaba**: AdWare:Win32/AdLoad.04d41639<br />**Jiangmin**: Backdoor.Wizard.j<br />**Kaspersky**: Trojan-Downloader.Win32.Adload.tsug<br />**Rising**: Downloader.Adload!8.D1 (CLOUD)<br />**Tencent**: Win32.Trojan-Downloader.Adload.Cgow |
| nuclei_2.8.8_windows_amd64.zip | 4024e624fed3d8896598cdb3872dd8cefef3989b8972bed4ef35a68c380e2405 | 3 / 63 | **Kaspersky**: Trojan-Downloader.Win32.Adload.tsug<br />**Tencent**: Win32.Trojan-Downloader.Adload.Umhl<br />**ZoneAlarm by Check Point**: Trojan-Downloader.Win32.Adload.tsug |
| nuclei_2.8.9_windows_386.exe | 7c75f0ee567fa86bc393eeb92ff4ac54496f51cb28e2dbb000dee9b44aea4aca | 1 / 69 | **Palo Alto Networks**: Generic.ml |
| nuclei_2.8.9_windows_386.zip | 706b8c17010a1948b450816a713fbcb4aa79aae5850fbb0dff4e0db03a0195dc | 0 / 64 | none |
| nuclei_2.8.9_windows_amd64.exe | 64a24860764bc9e4c15a8e450e76786e40e00b54448d5297242b8ca15bea028a | 0 / 68 | none |
| nuclei_2.8.9_windows_amd64.zip | 17b16681b5ff9fba4f22e1c78e87b89a96146a9652d8ac985e80761460785fdc | 0 / 61 | none |

**Conclusion and further thoughts**
After conducting an analysis on VirusTotal and a local machine running Windows 11, it has been concluded that there is nothing alarming emerging from the binaries. As a result, this event could be classified as a minor severity.

Most importantly, I think it is a good thing that nuclei binaries are flagged as potential dangerous software. This circumspects the abuse of nuclei on unauthorized hosts.

For legitimate purposes, the administrator can add the file to the Windows Defender's exclusion list. This has been suggested on the Microsoft forum [here](https://answers.microsoft.com/en-us/windows/forum/all/microsoft-defender-finds-trojan-but-in-virustotal/a79bd2b8-77e6-4e82-92f4-4d4f11b2c1c2).

I'm voting to close this issue as an **Informative** issue.

What are your thoughts?

Regards.

**Comment by toufik-airane** [comment]
Important information, I missed to specify that **there is a high probability that nuclei templates could be detected as malicious**, since many of these templates have an active payload in their content.

To operate nuclei templates in the best condition, you should definitely add the working folder in the list of Windows Defender's exclusion list.

Thank you.
